src/bucket.js

```diff
@@ -28,39 +29,50 @@ export default class Bucket {
      */
     this.name = name;
     /**
-     * The default options object.
      * @ignore
-     * @type {Object}
      */
-    this.options = options;
+    this._isBatch = !!options.batch;
     /**
      * @ignore
      */
-    this._isBatch = !!options.batch;
+    this._headers = options.headers || {};
+    this._retry = options.retry || 0;
+    this._safe = !!options.safe;
   }
 
   /**
-   * Merges passed request options with default bucket ones, if any.
+   * Get the value of "headers" for a given request, merging the
+   * per-request headers with our own "default" headers.
    *
    * @private
-   * @param  {Object} [options={}] The options to merge.
-   * @return {Object}              The merged options.
    */
-  _bucketOptions(options = {}) {
-    const headers = {
-      ...(this.options && this.options.headers),
-      ...options.headers,
-    };
+  _getHeaders(options) {
     return {
-      ...this.options,
-      ...options,
-      headers,
-      bucket: this.name,
-      batch: this._isBatch,
+      ...this._headers,
+      ...options.headers,
     };
   }
 
   /**
+   * Get the value of "safe" for a given request, using the
+   * per-request option if present or falling back to our default
+   * otherwise.
+   *
+   * @param {Object} options The options for a request.
+   * @returns {Boolean}
+   */
+  _getSafe(options) {
+    return getOptionWithDefault(options, "safe", this._safe);
+  }
+
+  /**
+   * As _getSafe, but for "retry".
+   */
```

Comment:
Please add a `@private` tag so these won't be documented as part of the public API (but maybe esdoc treats anything prefixed with _ as a private member already?)